test/integration/main_test.go

```diff
@@ -236,6 +256,100 @@ func forceUpdateSecret(ns string, secretName string, secret *v1.Secret) error {
 	return nil
 }
 
+func checkPodsLogs(t *testing.T, kubeClient *kubernetes.Clientset, message string) {
+	_checkPodsLogs(t, kubeClient, message, false)
+}
+
+func checkNewPodsLogs(t *testing.T, kubeClient *kubernetes.Clientset, message string) {
+	_checkPodsLogs(t, kubeClient, message, true)
+}
+
+func ExecCmd(t *testing.T, client kubernetes.Interface, podName string, namespace string,
+	command string, stdin io.Reader) (string, string, error) {
+	cmd := []string{
+		"/bin/bash",
+		"-c",
+		command,
+	}
+	req := client.CoreV1().RESTClient().Post().Resource("pods").Name(podName).
+		Namespace(namespace).SubResource("exec")
+	option := &corev1.PodExecOptions{
+		Command: cmd,
+		Stdin:   true,
+		Stdout:  true,
+		Stderr:  true,
+		TTY:     false,
+	}
+	if stdin == nil {
+		option.Stdin = false
+	}
+	req.VersionedParams(
+		option,
+		scheme.ParameterCodec,
+	)
+	exec, err := remotecommand.NewSPDYExecutor(kubeconfig(), "POST", req.URL())
+	if err != nil {
+		return "","", err
+	}
+	var stdout, stderr bytes.Buffer
+	err = exec.Stream(remotecommand.StreamOptions{
+		Stdin:  stdin,
+		Stdout: &stdout,
+		Stderr: &stderr,
+	})
+	if err != nil {
+		return "","",err
+	}
+
+	return stdout.String(), stderr.String(), nil
+}
+
+func e(t *testing.T, err error, message string) {
+	if err != nil {
+		t.Fatal(message, err.Error())
+	}
+}
+
+func findPod(t *testing.T, kubeClient *kubernetes.Clientset, namespace string, prefix string) *corev1.Pod {
+	newPods, err := kubeClient.CoreV1().Pods(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	for _, newPod := range newPods.Items {
+		if strings.HasPrefix(newPod.Name, prefix) {
+			return &newPod
+		}
+	}
+	return nil
+}
+
+func PodsLogsExist(t *testing.T, kubeClient *kubernetes.Clientset) bool {
```

Comment:
At first I was not exacly sure what this method does just by its name. Pod logs is quite specific, but its not related actually. What about calling it something like ArchiveContainsPodLogs, or anything else :)
Coudln't you also please take out code to degrate in separate function, maybe degradeMonitoring, also this would be both neede to be reverted in the end. For that I would call it alrady from TestPodLogsCollected.
You could use return defer function, something like:
```
func action() func() {
	fmt.Println("doing action")
	return func() {
		fmt.Println("reverting action")
	}
}

func main() {
	defer action()()

	fmt.Println("main")
}
```